pkg/istio-agent/sds-agent_test.go

```diff
@@ -1,39 +0,0 @@
-// Copyright Istio Authors
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package istioagent
-
-import (
-	"testing"
-
-	mesh "istio.io/api/mesh/v1alpha1"
-)
-
-// Validate that Agent comes up without errors when configured with file mounted certs.
```

Comment:
how come this is deleted? Seems like you deleted the code for this and then also deleted the test?

Reply:
The code is still there - but all env variables are moved to agent main. 

The test is very miss-leading - the agent would start up either way, and setting the env is not sufficient for proper operation ( even if the test was passing ) - nothing will work unless you have the files in /etc/certs

I'm using the 'externally managed' in the pilot gateway - when I bring it will also act as an integration test for the use case of 'someone creates certs in /etc/certs and istio-agent starts'

Reply:
> The test is very miss-leading - the agent would start up either way

This is not correct. Agent used to fail with an error. That is why this test was added. May be rename it to indicate that it just validates agent comes up as test says?

Reply:
IMO a wrong test can be worse than no test - it gives a false sense that something is tested.  I can add it back if you really want it - but if you look in the XDS package, we now have tests that run Pilot and the agent and test TLS connections work, 
including authentication. It's been a pain to get the infra in place - but now it should be much easier to extend them, and generating a static file+using this option is not very hard. 

This test was basically checking the agent class was not complaining if the setting was there - not that it was working. There is far more code involved that could go wrong. Worse: I couldn't even find a doc describing how it should work...